plugin/src/ast_helpers/number.rs

```diff
@@ -5,37 +5,81 @@ use tree_sitter_lint::{tree_sitter::Node, QueryMatchContext};
 
 use crate::{assert_kind, kind};
 
+#[derive(Copy, Clone, Debug)]
+pub enum NumberOrBigInt {
+    Number(Number),
+    BigInt(i64),
+}
+
 #[derive(Copy, Clone, Debug)]
 pub enum Number {
     NaN,
     Integer(i64),
     Float(f64),
 }
 
-impl Number {
+impl NumberOrBigInt {
     pub fn is_truthy(&self) -> bool {
         match self {
-            Number::NaN => false,
-            Number::Integer(value) => *value != 0,
-            Number::Float(value) => *value != 0.0,
+            Self::Number(Number::NaN) => false,
+            Self::Number(Number::Integer(value)) => *value != 0,
+            Self::Number(Number::Float(value)) => *value != 0.0,
+            Self::BigInt(value) => *value != 0,
+        }
+    }
+
+    pub fn eq_value(&self, other: &Self) -> bool {
```

Comment:
It seemed "correct" to have `PartialEq` implement the "stricter" equality-checking (where BigInt's != numbers), but for cases where you want the "looser" comparison/ordering, you can "opt in" to it via these `.eq_value()`/`.partial_cmp_value()` methods